infrastructure/simulation-tool/src/operations/deposit.ts

```diff
@@ -0,0 +1,119 @@
+import { Wallet as RollupWallet, RootstockOperation } from '@rsksmart/rif-rollup-js-sdk';
+import { PriorityOperationReceipt } from '@rsksmart/rif-rollup-js-sdk/build/types';
+import { BigNumber, constants, ContractReceipt, Wallet as EthersWallet } from 'ethers';
+import config from '../utils/config.utils';
+import { getRandomBigNumber } from '../utils/number.utils';
+import { chooseRandomWallet } from '../utils/wallet.utils';
+import { sleep } from '@rsksmart/rif-rollup-js-sdk/build/utils';
+
+type PreparedDeposit = Omit<Parameters<RollupWallet['depositToSyncFromRootstock']>[number], 'amount'> & {
+    amount: BigNumber;
+    from: RollupWallet;
+};
+
+type DepositResult = {
+    opL1Receipt: ContractReceipt;
+    opL2Receipt: PriorityOperationReceipt;
+    verifierReceipt: PriorityOperationReceipt;
+};
+
+const prepareDeposit = (l2sender: RollupWallet, l1recipient: EthersWallet, amount?: BigNumber): PreparedDeposit => {
+    const [minAmount, maxAmount] = config.weiLimits.deposit;
+
+    return {
+        amount: amount || getRandomBigNumber(minAmount, maxAmount),
+        depositTo: l1recipient.address,
+        token: constants.AddressZero,
+        from: l2sender
+    };
+};
+
+const executeDeposit = (preparedDeposit: PreparedDeposit): Promise<RootstockOperation> => {
+    const { from, ...depositParameters } = preparedDeposit;
+    console.log(`Depositing ${depositParameters.amount} RBTC to L2 ...`);
+
+    return from.depositToSyncFromRootstock(depositParameters);
+};
+
+const resolveRootstockOperation = async (rskOperation: RootstockOperation): Promise<DepositResult> => {
+    const opL1Receipt: ContractReceipt = await rskOperation.awaitRootstockTxCommit();
+    console.log(
+        `Rollup contract for RSK operation with hash #${rskOperation.ethTx.blockHash} ${
+            opL1Receipt.status ? 'submitted' : 'rejected'
+        }`
+    );
+    const opL2Receipt: PriorityOperationReceipt = await rskOperation.awaitReceipt();
+    console.log(
+        `Priority operation with hash #${rskOperation.ethTx.blockHash} ${
```

Comment:
I never the got the `blockHash`, its always undefined. I was able to get the `blockHash` by doing this.
```
const receipt = await rskOperation.ethTx.wait();
receipt.blockHash
```

Reply:
changed